rasa/core/tracker_store.py

```diff
@@ -468,6 +470,7 @@ def retrieve(self, sender_id):
                     sender_id, stored.get("events"), self.domain.slots
                 )
             else:
+                # should this be warnings.warn?
```

Comment:
Good point! I'm not sure how it could be `None` here unless it passed explicitly passed as such. Should I just add a type hint to make it `domain: Domain`?
Also the SQL tracker store does have `domain: Optional[Domain] = None`, is that correct?